src/openassetio-python/bridge/src/python/converter.cpp

```diff
@@ -44,7 +45,7 @@ PyObject* castToPyObject(const T& objectPtr) {
   const py::error_scope castErrorState;
 
   if (castErrorState.value != nullptr) {
-    throw std::runtime_error(py::str(castErrorState.value));
+    throw errors::InputValidationException(py::str(castErrorState.value));
```

Comment:
[plan] Docs need updating (and for below).

Reply:
Addressed as part of #1104